Tools/Scripts/webkitdirs.pm

```diff
@@ -2480,13 +2480,28 @@ sub isCachedArgumentfileOutOfDate($@)
     }
 
     open(CONTENTS_FILE, $filename);
+    local $/;
     chomp(my $previousContents = <CONTENTS_FILE> || "");
     close(CONTENTS_FILE);
 
     if ($previousContents ne $currentContents) {
-        print "Contents for file $filename have changed.\n";
-        print "Previous contents were: $previousContents\n\n";
-        print "New contents are: $currentContents\n";
+        print "Contents for file $filename have changed:\n";
+        # Print only lines that are different
+        my @new_lines = split /\n/, $currentContents;
+        my $i = 0;
+        for my $old_line (split /\n/, $previousContents) {
+            my $new_line = $new_lines[$i] // '';
+            if ($old_line ne $new_line) {
+                print "- $old_line\n";
+                print "+ $new_line\n";
+            }
+            $i++;
+        }
```

Comment:
This may break in subtle ways if the list of environment variables changes, wouldn't it be better to have the old file lines in a set, the same for the new file lines, and then if the sets are different, subtract/intersect them to show the differences? Then the order and amount of lines is irrelevant and the logic will always work correctly. Also, using sets would work when both files contain the same lines in different order.